src/main/scala/sbtbuildinfo/package.scala

```diff
@@ -4,23 +4,34 @@ package object sbtbuildinfo {
   type BuildInfoKey = BuildInfoKey.Entry[_]
   object BuildInfoKey {
     implicit def setting[A](key: SettingKey[A]): Entry[A] = Setting(key)
+    @deprecated("Explicitly wrap in BuildInfoKey.of/ofN. Or if out-of-graph execution is required use BuildInfoKey.outOfGraphTaskExecution", "0.7.1")
     implicit def task[A](key: TaskKey[A]): Entry[A] = Task(key)
+    implicit def taskValue[A: Manifest](task: sbt.Task[A]): Entry[A] = TaskValue(task)
     implicit def constant[A: Manifest](tuple: (String, A)): Entry[A] = Constant(tuple)
     
     def apply[A](key: SettingKey[A]): Entry[A] = Setting(key)
+    @deprecated("Explicitly wrap in BuildInfoKey.of/ofN. Or if out-of-graph execution is required use BuildInfoKey.outOfGraphTaskExecution", "0.7.1")
     def apply[A](key: TaskKey[A]): Entry[A] = Task(key)
     def apply[A: Manifest](tuple: (String, A)): Entry[A] = Constant(tuple)
     def map[A, B: Manifest](from: Entry[A])(fun: ((String, A)) => (String, B)): Entry[B] =
       BuildInfoKey.Mapped(from, fun)
     def action[A: Manifest](name: String)(fun: => A): Entry[A] = Action(name, () => fun)
 
+    def of(x: Any): BuildInfoKey = macro BuildInfoKeyMacros.ofImpl
+    def ofN(xs: Any*): Seq[BuildInfoKey] = macro BuildInfoKeyMacros.ofNImpl
+
+    def outOfGraphTaskExecution[A](key: TaskKey[A]): Entry[A] = Task(key)
```

Comment:
Maybe shorter name like `outOfGraph(..)`? Also is this tested ?

Reply:
I went with the more verbose name, to both make the intention explicit and to discourage usage.

It's tested via the identical deprecated implicit conversion. Is that ok? Or do you want me to switch to this one? Or do you want me to add both?

Reply:
@eed3si9n wdyt?

Reply:
`outOfGraphUnsafe`?